### DISCUSSION TITLE
Programmatically downloading extension

### DISCUSSION THREAD

**Discussion by Chad-ll** [redhat-developer/vscode-java#3696] [Q&A] [answered]
Hi everyone --

I support a bunch of developers on an air-gapped network that use this extension quite heavily, so I set up a process to automatically mirror the extension to one of my systems, then copy it to the air-gapped network.

I am trying to update my mirror script for the new file/URL format (java-linux-x64-$version-$number.vsix) but I can't figure out where the "$number" value is coming from at the end of the filename. 

- i.e.: for the current release of v1.33.0, the full filename is: java-linux-x64-1.33.0-**220**.vsix
- Where does that _220_ come from?

And more importantly, how can I figure out that $number in my script? I'm hoping that there is a file buried in the source tree that I didn't see that I can read and figure it out.

What I am doing now:

```
git_repo_root=${mirror}/other/vscode-repos
repo_root=${mirror}/other/vscode-vsix

devel=redhat-developer
repo=vscode-java

cd ${git_repo_root}/${repo}
git pull -a > /dev/null

#
# List all of the tags and get the most recent one that isn't a pre-release
git_version=$(git tag -l | sort -V | grep -Ev '(insider|prerelease|debug|preview)'| tail -1)
version=${git_version:1}

#
# Fetch the vsix files - We get Linux and Win64
#
for file in java-win32-x64 java-linux-x64 ; do
   url="https://github.com/${devel}/${repo}/releases/download/${git_version}/${file}-${version}.vsix"
   curl -L -s -o "${repo_root}/${file}.vsix" ${url}
done
```

Thank you

**Comment by fbricon** [comment, marked as answer]
The number comes from the `$GITHUB_RUN_NUMBER` value https://github.com/redhat-developer/vscode-java/blob/master/.github/workflows/release.yml#L142. It's quite unpredictable unfortunately.

But you should be able to identify the assets you need by using Github's API: https://docs.github.com/en/rest/releases/assets?apiVersion=2022-11-28#list-release-assets

You could also query the VS Code Marketplace API to do something similar, but I don't have pointers to give you.

**Reply by Chad-ll** [reply]
Hi Fred --

Thank you so much, the GitHub API gets me exactly what I'm looking for and solves the problem. Thank you for your help.

-chad
